raysort/s3_utils.py

```diff
@@ -176,11 +176,14 @@ def _upload_part(data):
         mpu_id, resp = mpu_queue.get()
         mpu_parts.append({"ETag": resp["ETag"], "PartNumber": mpu_id})
 
-    s3_client.complete_multipart_upload(
-        Bucket=pinfo.bucket,
-        Key=pinfo.path,
-        MultipartUpload={"Parts": mpu_parts},
-        UploadId=mpu["UploadId"],
-    )
-    pinfo.size = bytes_count
-    return [pinfo]
+    if len(mpu_parts) > 0:
+        s3_client.complete_multipart_upload(
+            Bucket=pinfo.bucket,
+            Key=pinfo.path,
+            MultipartUpload={"Parts": mpu_parts},
+            UploadId=mpu["UploadId"],
+        )
+        pinfo.size = bytes_count
+        return [pinfo]
+    else:
```

Comment:
`else` is redundant. Just `return [None]`

Can you also change this function's type signature to `list[Optional[PartInfo]]`? This also means `main_push.py:final_merge()`'s return value is now `Optional[PartInfo]`.